docs/design.md

```diff
@@ -1,317 +1,135 @@
-`ServiceBinding` resource:
-
-    kubectl create -f binding.yaml
-
-where `instance.yaml` might look like:
-
-    apiVersion: servicecatalog.k8s.io/v1beta1
-    kind: ServiceBinding
-    metadata:
-      namespace: example-ns
-      name: test-database-binding
-    spec:
-      instanceRef:
-        name: test-database
-      secretName: db-secret
-
-The Controller, upon being notified of the new `ServiceBinding` resource, will
-then talk to the Service Broker to create a new ServiceBinding for the specified
-Service Instance.
-
-Within the ServiceBinding object that is returned from the Service Broker are
-a set of Credentials. These Credentials contain all of the information
-needed for the application to talk with the Service Instance. For example,
-it might include things such as:
-- coordinates (URL) of the Service Instance
-- user-id and password to access the Service Instance
-
-The OSB API specification does not mandate what properties might appear
-in the Credentials, so the Application is required to understand the
-specified data returned and how to use it properly. This is typically done
-by reading the documentation of the Service.
-
-The Credentials will not be stored in the Service Catalog's datastore.
-Rather, they will be stored in the Kubenetes core as Secrets and a reference
-to the Secret will be saved within the `ServiceBinding` resource. If the
-ServiceBinding `spec.secretName` is not specified then the Controller will
-use the ServiceBinding `metadata.name` property as the name of the Secret.
-
-ServiceBindings are not required to be in the same Kubenetes Namespace
-as the Service Instance. This allows for sharing of Service Instances
-across Applications and Namespaces.
-
-Once the Secret is made available to the Application's Pods, it is then up
-to the Application code to use that information to talk to the Service
-ServiceInstance.
+The above is the high level architecture of Service Catalog.
 
-### Deleting Service Instances
+Service Catalog has two basic building blocks: an API server and a controller.
+This design is similar to the design of Kubernetes itself (in fact, 
+Service Catalog borrows a lot of code from Kubernetes to implement this
+design).
 
-As with all resources in Kubernetes, you can delete any of the Service
-Catalog resource by doing an HTTP DELETE to the resource's URL. However,
-it is important to note the you can not delete a Service Instance while
-there are ServiceBindings associated with it.  In other words, before a Service
-ServiceInstance can be deleted, you must first delete all of its ServiceBindings.
+### API Server
 
-Deleting a `ServiceBinding` indicates a user's intent to unbind.  The service-catalog controller handles deleting any Secrets associated with a binding.
+The API Server is an HTTPS
+[REST](https://en.wikipedia.org/wiki/Representational_state_transfer)ful 
+front-end for [etcd](https://coreos.com/etcd/) 
+(we can implement more storage backends, but we haven't done so)
```

Comment:
Is that even worth mentioning in the doc that we can implement other storage backends in the future?

Reply:
the possibility that we can do it is near and dear to my ❤️

Reply:
Fair enough. For me it just sounds a bit weird, as if we promised something without even planning to make it real :)